src/store/data.ts

```diff
@@ -51,9 +59,19 @@ export const initDataState = (
 /** Reset the data state after read/write is complete */
 export const clearDataState = () => {
   const data = currentData!;
+
   if (!data.gcScheduled && data.gcBatch.size > 0) {
     data.gcScheduled = true;
-    defer(() => gc(data));
+    defer(() => {
+      gc(data);
+    });
+  }
+
+  if (storage.current) {
+    defer(() => {
+      (storage.current as StorageAdapter).write(persistanceBatch);
```

Comment:
Small typo on “persistence”, also I don’t think we need the ref structure anymore :)

Reply:
how would you set the storageAdapter from the exchange then? Since we are explicitly calling this